pkg/crypto/rings/interface.go

```diff
@@ -0,0 +1,15 @@
+package rings
+
+import (
+	"context"
+
+	"github.com/noot/ring-go"
+)
+
+// RingCache is used to store rings used for signing and verifying relay requests.
+// It will cache rings for future use after querying the application module for
+// the addresses of the gateways the application is delegated to, and converting
```

Comment:
```suggestion
// the pubKeys of the gateways the application is delegated to, and converting
```

Reply:
We are actually querying for addresses tho, thats how we store them in the application